django/middleware/csrf.py

```diff
@@ -105,19 +105,38 @@ def _reject(self, request, reason):
         )
         return _get_failure_view()(request, reason=reason)
 
+    def _get_current_token(self, request):
+        try:
+            if settings.CSRF_USE_SESSIONS:
+                return _sanitize_token(request.session[settings.CSRF_COOKIE_NAME])
```

Comment:
I agree that using settings.CSRF_COOKIE_NAME here is not elegant, but neither is setting it to a hardcoded value and introducing more settings bad as well; but I'm open to discussion ok this.